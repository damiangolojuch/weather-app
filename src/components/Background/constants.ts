export const backgrounds = [
  "pexels-andy-vu-3244513.jpg",
  "pexels-andy-vu-3484061.jpg",
  "pexels-asad-photo-maldives-1591373.jpg",
  "pexels-frans-van-heerden-624015.jpg",
  "pexels-freestocksorg-839462.jpg",
  "pexels-lumn-167699.jpg",
  "pexels-pixabay-235621.jpg",
  "pexels-roberto-nickson-2559941.jpg",
  "pexels-roberto-nickson-2775196.jpg",
  "pexels-stein-egil-liland-3408744.jpg",
];
