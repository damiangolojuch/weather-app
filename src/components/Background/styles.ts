import styled from "styled-components";
import config from "~/styles/config";

export const BackgroundWrapper = styled.div`
  width: 100%;
  height: 100%;
  background-color: ${config.colors.main};
  overflow: hidden;
`;

export const Background = styled.img`
  display: block;
  width: 100%;
  height: 100%;
  opacity: 0.6;
`;
