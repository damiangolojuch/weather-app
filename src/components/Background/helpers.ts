import { backgrounds } from "~/components/Background/constants";

export const getRandomBackground = () =>
  backgrounds[Math.floor(Math.random() * backgrounds.length)];
