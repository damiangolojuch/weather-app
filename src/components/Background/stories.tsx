import { Story, Meta } from "@storybook/react";
import Background from ".";

export default {
  title: "Background",
  component: Background,
  args: {
    description: "Background component with random image",
  },
} as Meta;

export const Default: Story = (args) => <Background {...args} />;
