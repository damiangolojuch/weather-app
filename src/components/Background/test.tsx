import Background from "./";
import { render, screen } from "@testing-library/react";

jest.mock("components/Background/helpers", () => {
  return {
    __esModule: true,
    getRandomBackground: function Mock() {
      return "random-img-path.jpg";
    },
  };
});

describe("<Background />", () => {
  it("should display img with image from backgrounds directory", () => {
    const { container } = render(<Background />);

    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(container.getElementsByTagName("img")).toMatchSnapshot();
  });
});
