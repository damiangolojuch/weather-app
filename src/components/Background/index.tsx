import * as S from "./styles";
import { FC, useMemo } from "react";
import { getRandomBackground } from "~/components/Background/helpers";

interface BackgroundProps {
  className?: string;
}

const Background: FC<BackgroundProps> = ({ className }) => {
  const image = useMemo(() => getRandomBackground(), []);
  const imgSrc = `${process.env.PUBLIC_URL || ""}/assets/backgrounds/${image}`;

  return (
    <S.BackgroundWrapper className={className}>
      <S.Background src={imgSrc} alt="Na licencji pexels.com" />
    </S.BackgroundWrapper>
  );
};

export default Background;
