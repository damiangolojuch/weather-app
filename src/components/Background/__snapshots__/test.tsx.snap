// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> should display img with image from backgrounds directory 1`] = `
HTMLCollection [
  <img
    alt="Na licencji pexels.com"
    class="sc-gsnTZi yRSuN"
    src="/assets/backgrounds/random-img-path.jpg"
  />,
]
`;
